src/utils/searchUtils.ts

```diff
@@ -21,3 +21,47 @@ export function isArNSDomainNameAvailable({
   }
   return true;
 }
+
+export function calculateArNSNamePrice({
+  domain,
+  years,
+  selectedTier,
+  fees,
+}: {
+  domain?: string;
+  years: number;
+  selectedTier: number | string;
+  fees: { [x: number]: number };
+}) {
+  const thisTier = new Number(selectedTier).valueOf();
+  try {
+    if (years < 1) {
+      throw Error('Minimum duration must be at least one year');
+    }
+    if (thisTier < 1) {
+      throw Error('Minimum tier is 1');
+    }
+    if (thisTier > 3) {
+      throw Error('Maximum tier is 3');
+    }
+    if (!domain) {
+      throw Error('Domain is undefined');
+    }
+    if (!isArNSDomainNameValid({ name: domain })) {
+      throw Error('invalid name, failed name regex');
+    }
+    if (isArNSDomainNameValid({ name: domain }) && domain) {
+      let nameLength = domain.length;
+      if (nameLength > Object.keys(fees).length) {
+        nameLength = Object.keys(fees).length;
+      }
+      const namePrice = fees[nameLength];
+      const price = namePrice * years * thisTier;
+
+      return price;
+    }
```

Comment:
ah, wasn't clear. i meant throw an error, but avoid the use of `else.
```suggestion
    if (!isArNSDomainNameValid({ name: domain })) {
      throw Error('Domain name is invalid');
    }
      const nameLength = Math.max(domain.length, Object.keys(fees).length);
      const namePrice = fees[nameLength];
      const price = namePrice * years * thisTier;

      return price;
    }
```